frontend/src/Components/OAuth2AppCardModalComponent.tsx

```diff
@@ -146,7 +157,17 @@ export function OAuth2AppCardModalComponent(props: OAuth2AppCardModalProps) {
 
   const handleInstallClick = () => {
     if (client.initialAuthorizationUrl) {
-      window.open(client.initialAuthorizationUrl, '_blank')
+      const authWindow = window.open(client.initialAuthorizationUrl, '_blank')
+      if (authWindow) {
+        const handleFocus = () => {
+          window.removeEventListener('focus', handleFocus)
+          api.oauth2Api.clearClientCache()
+          api.oauth2Api.getClientCached(client.clientId).then((updatedClient) => {
+            props.onClientUpdate?.(updatedClient)
+          })
+        }
+        window.addEventListener('focus', handleFocus)
```

Comment:
I think it can be something like this
```js
window.addEventListener('focus', () => {
}, { once: true })
```
and then you won't need to extract the callback and call remove listener